packages/kbn-expandable-flyout/src/provider.tsx

```diff
@@ -9,21 +9,20 @@
 import { createKbnUrlStateStorage } from '@kbn/kibana-utils-plugin/public';
 import React, { FC, PropsWithChildren, useEffect, useMemo } from 'react';
 import { Provider as ReduxProvider } from 'react-redux';
-
 import { useHistory } from 'react-router-dom';
-import { State } from './state';
+import { ExpandableFlyoutContextProvider, useExpandableFlyoutContext } from './context';
+import { FlyoutState } from './state';
 import { useExpandableFlyoutState } from './hooks/use_expandable_flyout_state';
-import { EXPANDABLE_FLYOUT_URL_KEY } from './constants';
-import { Context, store, useDispatch } from './redux';
+import { Context, selectNeedsSync, store, useDispatch, useSelector } from './redux';
 import { urlChangedAction } from './actions';
 
-export type ExpandableFlyoutStorageMode = 'memory' | 'url';
-
 /**
  * Dispatches actions when url state changes and initializes the state when the app is loaded with flyout url parameters
  */
 const UrlSynchronizer = () => {
```

Comment:
Additionaly, there are tests lacking for this file completely. Could you please add those?

Reply:
I don't think we can change this to a hook because we need to conditionally render it (look line 96 of the file).
I can extract it though and write some tests for it!

Reply:
I finally decided against extracting it, but I have added unit tests

Reply:
>I don't think we can change this to a hook because we need to conditionally render it (look line 96 of the file).

I think that is more of a design question. It can simply be made into a hook by passing `needSync : boolean` as a parameter to `useUrlSync` and then run effects based on that. I still think that it is more suitable as hook. 

But It is only `nit` from my side so please go ahead with what you think is good. Thanks for making other changes.

Reply:
I'm still not sure how I would be able to do this. Let me ping on on Slack